Create header

bxxmi opened the pull request.

bxxmi committed:
docs : Update README.md content

write first mini project topic

bxxmi committed:
docs: Reupdate README

Reupdate README.md

bxxmi committed:
feat: Create GitHub main header

create github main page header from left menus

bxxmi committed:
feat: Applied github main header CSS

applied github main page header CSS

heejinlee07 commented on index.html:
```diff
@@ -0,0 +1,41 @@
+<!DOCTYPE html>
+<html lang="ko">
+<head>
+    <meta charset="UTF-8">
+    <title>GitHub</title>
+    <link rel="stylesheet" href="./index.css">
```
<Head>영역에 favicon을 추가하면 좋습니다. 일반적으로 모든 사이트에서는 favicon이 사용되고 있습니다. 자세한 사용방법은 아래 링크의 글을 읽어보시기 바랍니다. favicon외에도 <Head>영역에 필수로 사용되는 태그들에 대한 정보가 있습니다.

[https://developer.mozilla.org/ko/docs/Learn/HTML/Introduction_to_HTML/The_head_metadata_in_HTML#맞춤_아이콘_추가하기](https://developer.mozilla.org/ko/docs/Learn/HTML/Introduction_to_HTML/The_head_metadata_in_HTML#%EB%A7%9E%EC%B6%A4_%EC%95%84%EC%9D%B4%EC%BD%98_%EC%B6%94%EA%B0%80%ED%95%98%EA%B8%B0)

bxxmi replied:
자세하게 알려주셔서 감사합니다!

heejinlee07 commented on index.html:
```diff
@@ -0,0 +1,41 @@
+            <!--왼쪽 메뉴를 감싸는 div-->
+            <div class="container-left clearfix">
+                <div class="logo">
+                    <img src="https://heropcode.github.io/GitHub-Responsive/img/logo.svg" alt="GitHub Logo">
```
이미지 태그에 대체텍스트를 작성하신 점이 좋아요. SEO 최적화를 고려하여 항상 대체텍스트를 작성하는 습관을 들여주세요:)

bxxmi replied:
네! 감사합니다!

heejinlee07 commented on index.html:
```diff
@@ -0,0 +1,41 @@
+                    <input type="text" name="search" placeholder="Search or jump to ...">
+                </div>
+                <!--실제 메뉴를 감싸는 div-->
+                <div class="menu">
+                    <div class="menu-item">Pull requests</div>
+                    <div class="menu-item">Issues</div>
+                    <div class="menu-item">Marketplace</div>
+                    <div class="menu-item">Explore</div>
+                </div>
```
화면을 구성할 때 시멘틱 태그를 고려하시면서 코드를 작성하는 습관을 들이시는 것이 좋습니다. 따라서 `<div>`로 작성하셔도 되지만 `<ul>`요소 안에` <li>`가 있는 구조로 작성하는 것이 시멘틱하게 코드를 구성하는 것입니다. 관련된 내용은 아래 링크를 참조하세요:)
https://snusang.tistory.com/4

bxxmi replied:
자세하게 말씀해주셔서 감사합니다!

heejinlee07 commented on index.css:
```diff
@@ -0,0 +1,64 @@
+    background-color: #24292E;
+}
+
+.container {
+    width: 1500px;
+    margin: auto;
+}
+
+.container-left {
+    color: white;
+    padding: 20px;
+}
+
+.logo {
+    float: left;
+    margin-right: 8px;
+}
+
+.logo img {
+    display: block;
+}
+
+.inputbox {
+    float:left;
+    padding:8px 20px;
+}
```
float는 현업에서 거의 쓰지 않는 속성입니다. float를 쓰지 않는 이유는 기존의 요소의 정렬을 깨트리고, 잘못 사용할 시 여러 사이드 이펙트를 일으킬 가능성이 높기 때문입니다. 무엇보다도 float를 대체할 수 있는 flexbox와 grid가 등장했기 때문에 float는 쓰지 않습니다. flexbox를 통해 화면을 구성하는 방법을 공부해보시면 좋겠습니다.

관련하여 읽어보면 좋은 링크는 아래와 같습니다.
https://blog.eunsatio.io/develop/%EB%A0%88%EC%9D%B4%EC%95%84%EC%9B%83-%EC%A0%95%EB%A0%AC%EC%9D%84-%EC%9C%84%ED%95%B4-css-float%EB%A5%BC-%EB%90%98%EB%8F%84%EB%A1%9D-%EC%93%B0%EC%A7%80-%EC%95%8A%EC%95%84%EC%95%BC-%ED%95%98%EB%8A%94-%EC%9D%B4%EC%9C%A0

flexbox를 쉽고 재밌게 공부할 수 있는 사이트입니다.
https://flexboxfroggy.com/

bxxmi replied:
감사합니다! flexbox가 있다는 것을 몰랐네요! 반영해보도록 하겠습니다!

heejinlee07 replied:
LGTM👍

heejinlee07 reviewed (commented):
안녕하세요 반갑습니다 😃
첫 리뷰를 작성하시느라 고생많으셨습니다.
flexbox 사용법에 대해 고민해보시면 좋겠습니다.
그리고, 현업에서는 보통 Conversation 란에 개발한 내용에 대해 기록하여 pr을 올리기 때문에
Conversation 란에 자신이 작업한 내용을 상세히 작성해주시면 좋습니다:)

bxxmi committed:
refactor: 시멘틱 태그를 활용한 nav 구현

기존에 <div>로 구현한 네비게이션을 시맨틱 태그인 <nav>와 <ul>,<li>를 활용해서 구현했습니다.

bxxmi committed:
refactor: float 정렬에서 flexbox로 정렬하기

기존 정렬 방식이었던 float 속성 말고 flexbox를 활용해서 변경하였고 로고, inputbox, 메인 메뉴들이 제대로 일자로 보이게 변경했습니다.

bxxmi committed:
Merge branch 'main' into create-header
